api-tests/src/test/java/org/openmrs/module/reporting/test/OpenmrsVersionTestListener.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.reporting.test;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assume;
+import org.openmrs.module.ModuleUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.springframework.test.context.TestContext;
+import org.springframework.test.context.support.AbstractTestExecutionListener;
+
+public class OpenmrsVersionTestListener extends AbstractTestExecutionListener {
```

Comment:
This allows us to skip tests based on the OpenMRS version with the `RequiresVersion` annotation.

Reply:
Very nice.